lib/kafka/cluster.rb

```diff
@@ -119,30 +119,48 @@ def partitions_for(topic)
       raise
     end
 
-    def resolve_offset(topic, partition, offset)
+    def resolve_offsets(topic, partitions, offset)
       add_target_topics([topic])
       refresh_metadata_if_necessary!
-      broker = get_leader(topic, partition)
+
+      partitions_by_broker = partitions.inject({}) {|hsh, partition|
+        broker = get_leader(topic, partition)
+        hsh[broker] ||= []
+        hsh[broker] << partition
+        hsh
+      }
```

Comment:
I’m wondering if this code would be clearer by using `each_with_object` and `Hash.new([])`?